api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -2027,11 +2027,14 @@ public static String getRuntimePropertiesFilePathName(String applicationName) {
 		
 		// first look in the current directory (that java was started from)
 		String pathName = fileNameInTestMode != null ? fileNameInTestMode : defaultFileName;
+		
 		log.debug("Attempting to look for properties file in current directory: " + pathName);
+		
+		List<String> errorMessages = new ArrayList<>();
 		if (new File(pathName).exists()) {
 			return pathName;
 		} else {
-			log.warn("Unable to find a runtime properties file at " + new File(pathName).getAbsolutePath());
+			errorMessages.add("Unable to find a runtime properties file at " + new File(pathName).getAbsolutePath());
```

Comment:
Here, etc.

```suggestion
            if (log.isWarnEnabled()) {
			     errorMessages.add("Unable to find a runtime properties file at " + new File(pathName).getAbsolutePath());
			}
```

That way we can avoid any overhead from this if the WARN level is disabled for this logger.